website/project/model.py

```diff
@@ -316,6 +352,12 @@ def edit(self, content, auth, save=False):
         self.content = content
         self.modified = True
         self.date_modified = datetime.datetime.utcnow()
+        new_mentions = get_valid_mentioned_users_guids(self, self.node.contributors)
+
+        if new_mentions:
+            project_signals.mention_added.send(self, new_mentions=new_mentions, auth=auth)
+            self.ever_mentioned.extend(new_mentions)
+
         if save:
```

Comment:
Picking up from https://github.com/CenterForOpenScience/osf.io/pull/5518/files/4a7237c12108b39f47b62dc169e8734e905891f0#r66857195 ...

> Right, `ever_mentioned` should only be modified if the comment is saved because mentioned users should only get notified if the comment is saved.

Here in the `edit` case we're sending notifications and extending `ever_mentioned` even if the comment is not saved. One way to clean this up would be to drop the `save` flag. Are we sure we're actually using that anywhere outside of tests?